das/daser_test.go

```diff
@@ -4,29 +4,254 @@ import (
 	"context"
 	"sync"
 	"testing"
+	"time"
+
+	"github.com/ipfs/go-datastore"
+	ds_sync "github.com/ipfs/go-datastore/sync"
+	format "github.com/ipfs/go-ipld-format"
+	mdutils "github.com/ipfs/go-merkledag/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 
 	"github.com/celestiaorg/celestia-node/service/header"
 	"github.com/celestiaorg/celestia-node/service/share"
 )
 
-func TestDASer(t *testing.T) {
-	shareServ, dah := share.RandLightServiceWithSquare(t, 16)
+// TestDASerLifecycle tests to ensure every mock block is DASed and
+// the DASer checkpoint is updated to network head.
+func TestDASerLifecycle(t *testing.T) {
+	ds := ds_sync.MutexWrap(datastore.NewMapDatastore())
+	cstore := wrapCheckpointStore(ds)
```

Comment:
Do you need to wrap it here if its wrapper inside the constructor

Reply:
resolved?